src/qx/QuantumState.cpp

```diff
@@ -13,23 +13,31 @@ void QuantumState::checkQuantumState() {
         throw QuantumStateError{ fmt::format("number of qubits exceeds maximum allowed: {} > {}", numberOfQubits,
                                              config::MAX_QUBIT_NUMBER) };
     }
+    if (numberOfBits > config::MAX_BIT_NUMBER) {
+        throw QuantumStateError{ fmt::format("number of bits exceeds maximum allowed: {} > {}", numberOfBits,
```

Comment:
Why us the number of bits relevant to the QuantumState? Won't the number of qubits exceed the maximum allowed, way before the number of bits exceeds its maximum?

Reply:
A `BitRegister` is a type of `Register`, and a register has a given size.
QX simulator uses a `config::MAX_QUBIT_NUMBER` because, for the way it operates (based in matrices, and matrices sizes being related to the number of qubits), it cannot manage beyond a given number of qubits (64).
Likewise, the `config::MAX_BIT_NUMBER` is just set for sanity to 1MB.

> Why us the number of bits relevant to the QuantumState?

#CQT-130 is a refactoring task that should address this. I think a quantum state and the measurements should be different things. They may well go  together in a simulation result, though.

> Won't the number of qubits exceed the maximum allowed, way before the number of bits exceeds its maximum?

They are separated things. You can have a system of 30 qubits and no bits (no measurements), or you could have a system of 1 qubit and 1MB bits.